src/test/integration/integrationtests.py

```diff
@@ -0,0 +1,122 @@
+import argparse
+import json
+import ssl
+import unittest
+import warnings
+
+import requests
+import urllib3
+
+try:
+    import urllib3.contrib.pyopenssl
+    urllib3.contrib.pyopenssl.inject_into_urllib3()
+except ImportError:
+    pass
+
+def read_config(config_path):
+    with open(config_path) as f:
+        return json.load(f)
+
+class CourseSubjectTests(unittest.TestCase):
+    config = None
+    _access_token = None
+
+    @property
+    def url(self):
+        return self.config['hostname'] + self.config['version'] + self.config['api']
+
+    def access_token(self):
+        if self._access_token is None:
+            url = self.config['hostname'] + self.config['version'] + self.config["token_endpoint"]
+            post_data = {
+                'client_id': self.config["client_id"],
+                'client_secret': self.config["client_secret"],
+                'grant_type': 'client_credentials',
+            }
+            response = requests.post(url, data=post_data)
+            self.assertEqual(response.status_code, 200)
+            self.assertEqual(response.headers['content-type'], 'application/json')
+            body = response.json()
+            self.assertEqual(body['token_type'], 'BearerToken') # XXX should be Bearer
+            self._access_token = body["access_token"]
+        return self._access_token
+
+    def authorization(self):
+        return 'Bearer ' + self.access_token()
+
+    def query(self):
+        headers = {'Authorization': self.authorization()}
+        return requests.get(self.url, headers=headers)
+
+    def check_ssl(self, protocol, url):
+        manager = urllib3.poolmanager.PoolManager(
+            ssl_version=protocol)
+        with warnings.catch_warnings():
+            warnings.simplefilter('ignore', urllib3.exceptions.InsecureRequestWarning)
+            try:
+                manager.request('GET', url)
+            except urllib3.exceptions.SSLError:
+                return False
+            else:
+                return True
+
+    def test_success(self):
+        """a request return 200 Success and a json object"""
+        response = self.query()
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.headers['content-type'], 'application/json')
+
+        with open('golden.json') as f:
+            golden = json.load(f)
+
+        body = response.json()
+        self.assertEqual(body, golden)
```

Comment:
We could check to see if there are specific entries present in the json such as the beginning of time and end of time terms :) .